Classify AlternateDataStream and test for gufo:Type instantiations among exemplars

ajnelson-nist opened the pull request.

ajnelson-nist committed:
Classify AlternateDataStream and test for gufo:Type instantiations among exemplars

No effects were observed on Make-managed files.

References:
* https://github.com/ucoProject/UCO/issues/590

Signed-off-by: Alex Nelson <alexander.nelson@nist.gov>

plbt5 commented on tests/exemplars.ttl:
```diff
@@ -29,6 +29,15 @@ kb:Action-13c91d40-9ca0-4558-81a1-4f7db888c371
 
+kb:AlternateDataStream-e6ceee40-5614-4740-b08c-d21ce9b2356d
+	a
+		uco-observable:AlternateDataStream ,
```
How can something be typed as both a gufo:Category (non-sortal) and a gufo:Kind (sortal). I think that is incorrect.

ajnelson-nist replied:
The reason both of these types are here is because `FileSystemObject` is going to be entailed as the superclass of `AlternateDataStream` after Issue 590 is incorporated.

More generally, my understanding is that it is fine for an individual thing T to be classified as in a class C1 that is a `gufo:Kind` AND as in a class C2 that is a `gufo:Category` that happens to be the parent of C1.  Doesn't this happen basically everywhere?

ajnelson-nist replied:
Does this illustration help?

```mermaid
flowchart BT

subgraph TBox
  _uco_obs_ads[uco-observable:AlternateDataStream]
  _uco_obs_fso[uco-observable:FileSystemObject]

  subgraph _gufo_types [gUFO Typing]
    _gufo_category[gufo:Category]
    _gufo_kind[gufo:Kind]
  end
end

subgraph ABox
  _kb_ads_e6[kb:AlternateDataStream-e6ceee40...]
end

_kb_ads_e6 -- ∈ --> _uco_obs_ads
_uco_obs_ads -- ⊂ --> _uco_obs_fso
_uco_obs_ads -- ∈ --> _gufo_kind
_uco_obs_fso -- ∈ --> _gufo_category

_gufo_kind -- ⋂=∅ --> _gufo_category
```

plbt5 replied:
What sets/classes do you refer to with the intersection in "⋂=∅"?

ajnelson-nist replied:
`gufo:Category` and `gufo:Kind` are disjoint, via disjointedness amongst their parent classes.

plbt5 replied:
Indeed, `gufo:Category` and `gufo:Kind` are disjoint. 
The disjointness doesn't create a conflict for `uco-observable:AlternateDataStream`, since the `gufo:Category` does not provide a principle of identity as `gufo:Kind` does.

ajnelson-nist committed:
Merge branch 'main' into classify_ntfs_ads

ajnelson-nist committed:
Merge branch 'main' into classify_ntfs_ads

ajnelson-nist committed:
Align AlternateDataStream as a subclass of gufo:Object

No effects were observed on Make-managed files.

Reported-by: Paul Brandt <paul@brandt.name>
Signed-off-by: Alex Nelson <alexander.nelson@nist.gov>

ajnelson-nist commented:
The argument for `AlternateDataStream` being a `gufo:Kind`, and `FileSystemObject` being a `gufo:Category`, stems from a sibling class: `File`, another subclass of `FileSystemObject`, has a different notion of identity from `AlternateDataStream`.
